ros_ws/src/dataset_generation/include/dataset_generation/filter_point_clouds.h

```diff
@@ -44,6 +45,24 @@ class PointCloudFiltering
         const float& leaf_size_x,
         const float& leaf_size_y,
         const float& leaf_size_z);
+    
+    pcl::PointCloud<pcl::PointXYZ>::Ptr apply_pass_filter(
```

Comment:
I think you should rename this to `remove_points_on_ground(...)`. Mostly because that is what it is doing specifically. A `pass_filter` could be doing a lot of other things, but this function specifically removes ground, so I think we should call it that. 

Same goes for the other two functions as well. You can call them `remove_ground_and_visualize()`, `remove_ground_and_save()`.